esphome/components/ld2420/number/__init__.py

```diff
@@ -0,0 +1,183 @@
+LD2420GateSelectNumber = ld2420_ns.class_("LD2420GateSelectNumber", number.Number)
+LD2420MoveThresholdNumbers = ld2420_ns.class_(
+    "LD2420MoveThresholdNumbers", number.Number
+)
+LD2420StillThresholdNumbers = ld2420_ns.class_(
+    "LD2420StillThresholdNumbers", number.Number
+)
+CONF_MIN_GATE_DISTANCE = "min_gate_distance"
+CONF_MAX_GATE_DISTANCE = "max_gate_distance"
+CONF_STILL_THRESHOLD = "still_threshold"
+CONF_MOVE_THRESHOLD = "move_threshold"
+CONF_GATE_MOVE_SENSITIVITY = "gate_move_sensitivity"
+CONF_GATE_STILL_SENSITIVITY = "gate_still_sensitivity"
+CONF_GATE_SELECT = "gate_select"
+CONF_PRESENCE_TIMEOUT = "presence_timeout"
+GATE_GROUP = "gate_group"
+TIMEOUT_GROUP = "timeout_group"
+
+
+CONFIG_SCHEMA = cv.Schema(
+    {
+        cv.GenerateID(CONF_LD2420_ID): cv.use_id(LD2420Component),
+        cv.Inclusive(CONF_PRESENCE_TIMEOUT, TIMEOUT_GROUP): number.number_schema(
+            LD2420TimeoutNumber,
+            unit_of_measurement=UNIT_SECOND,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon=ICON_TIMELAPSE,
+        ),
+        cv.Inclusive(CONF_MIN_GATE_DISTANCE, TIMEOUT_GROUP): number.number_schema(
+            LD2420MinDistanceNumber,
+            device_class=DEVICE_CLASS_DISTANCE,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon=ICON_MOTION_SENSOR,
+        ),
+        cv.Inclusive(CONF_MAX_GATE_DISTANCE, TIMEOUT_GROUP): number.number_schema(
+            LD2420MaxDistanceNumber,
+            device_class=DEVICE_CLASS_DISTANCE,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon=ICON_MOTION_SENSOR,
+        ),
+        cv.Inclusive(CONF_GATE_SELECT, GATE_GROUP): number.number_schema(
+            LD2420GateSelectNumber,
+            device_class=DEVICE_CLASS_DISTANCE,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon=ICON_MOTION_SENSOR,
+        ),
+        cv.Inclusive(CONF_STILL_THRESHOLD, GATE_GROUP): number.number_schema(
+            LD2420StillThresholdNumbers,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon=ICON_MOTION_SENSOR,
+        ),
+        cv.Inclusive(CONF_MOVE_THRESHOLD, GATE_GROUP): number.number_schema(
+            LD2420MoveThresholdNumbers,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon=ICON_MOTION_SENSOR,
+        ),
+        cv.Optional(CONF_GATE_MOVE_SENSITIVITY): number.number_schema(
+            LD2420MoveSensFactorNumber,
+            device_class=DEVICE_CLASS_DISTANCE,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon=ICON_SCALE,
+        ),
+        cv.Optional(CONF_GATE_STILL_SENSITIVITY): number.number_schema(
+            LD2420StillSensFactorNumber,
+            device_class=DEVICE_CLASS_DISTANCE,
+            entity_category=ENTITY_CATEGORY_CONFIG,
+            icon=ICON_SCALE,
+        ),
+    }
+)
+CONFIG_SCHEMA = CONFIG_SCHEMA.extend(
+    {
+        cv.Optional(f"g{x}"): (
```

Comment:
Quick thought -- since we use the full word "gate" in the other parameters, could/should we use it here, too? I think maybe like this:
```suggestion
        cv.Optional(f"gate_{x}"): (
```

Reply:
For sure. Added change.